code/hdb/startup.q

```diff
@@ -1,4 +1,55 @@
 
+//called after new HDB cluster finishes loading all dependencies
+//triggers RDB reload instead of the HDB and removes old HDB cluster if it is a replacement
+.hdb.startup:{
+  gws:exec w from .servers.SERVERS where proctype = `gateway, .dotz.liveh w;
+  if[not count gws; :()]; // TODO : Handle case when none
+
+  // block queries to new hdb
+  gws @\: (`.gw.setserveractiveflag;.proc.procname;0b);
+  // wait for status change
+  .finspace.checkstatus[(.finspace.getcluster;.proc.procname);("RUNNING";"FAILURE");00:01;0D01];
+
+  // identify the row and retry connection
+  {[h;tgt]
+     tgtidx:first @[h;({exec i from .servers.SERVERS where procname=x};tgt);()];
+     if[null tgtidx; :()];
+     doretry:@[h;({null .servers.SERVERS[x][`w]};tgtidx);1b];
+     .lg.o[`hdbstartup;"will force retry row ",(-3!tgtidx)," in servers.SERVERS for gw handle ",(-3!h)," : ",-3!doretry];
+     if[doretry; h(`.servers.retryrows;tgtidx)];
+   }[;.proc.procname] each gws;
```

Comment:
So I guess we are opening connections from HDB->GW and GW->HDB, and in this piece of code we're using the HDB->GW connection to tell GW to retry the GW->HDB connection if needed? I think the comment above this section is a bit vague and could be more specific

Reply:
yes, because the GW->HDB connection may not successfully open when the HDB is still in a "CREATING" state